heron/spi/src/java/com/twitter/heron/spi/statemgr/IStateManager.java

```diff
@@ -276,4 +277,27 @@ public String getName() {
    */
   ListenableFuture<PhysicalPlans.PhysicalPlan> getPhysicalPlan(
       WatchCallback watcher, String topologyName);
+
+  /**
+   * Set the Stateful Checkpoint
+   *
+   * @return Boolean - Success or Failure
+   */
+  ListenableFuture<Boolean> setStatefulCheckpoint(
+      CheckpointManager.StatefulConsistentCheckpoints checkpoint, String topologyName);
```

Comment:
This method should set a single checkpoint, so `StatefulConsistentCheckpoint` not `StatefulConsistentCheckpoints`. It will also need to set them in a way that we can a.) delete a single checkpoint and b.) identify the packing plan that it's bound to.